sec_certs/sample/fips.py

```diff
@@ -1,206 +1,202 @@
+        divs = self._soup.find_all("div", class_="panel panel-default")
+        details_div, vendor_div, related_files_div, validation_history_div = divs
+        details_dict = self._build_details_dict(details_div)
 
-    @staticmethod
-    def parse_vendor(current_div: Tag, html_items_found: Dict, current_file: Path) -> None:
-        vendor_string = current_div.find("div", "panel-body").find("a")
+        vendor_dict = self._build_vendor_dict(vendor_div)
+        related_files_dict = self._build_related_files_dict(related_files_div)
+        validation_history_dict = self._build_validation_history_dict(validation_history_div)
 
-        if not vendor_string:
-            vendor_string = list(current_div.find("div", "panel-body").children)[0].strip()
-            html_items_found["vendor_www"] = ""
+        if "algorithms" in details_dict:
+            algorithms_data = details_dict.pop("algorithms")
+            algorithms = set()
+            for category, alg_ids in algorithms_data.items():
+                algorithms |= {category + x for x in alg_ids}
         else:
-            html_items_found["vendor_www"] = vendor_string.get("href")
-            vendor_string = vendor_string.text.strip()
+            algorithms = set()
 
-        html_items_found["vendor"] = vendor_string
-        if html_items_found["vendor"] == "":
-            logger.warning(f"NO VENDOR FOUND {current_file}")
-
-    @staticmethod
-    def parse_lab(current_div: Tag, html_items_found: Dict, current_file: Path) -> None:
-        html_items_found["lab"] = list(current_div.find("div", "panel-body").children)[0].strip()
-        html_items_found["nvlap_code"] = (
-            list(current_div.find("div", "panel-body").children)[2].strip().split("\n")[1].strip()
+        return algorithms, FIPSCertificate.WebData(
+            **{**details_dict, **vendor_dict, **related_files_dict, **validation_history_dict}
         )
 
-        if html_items_found["lab"] == "":
-            logger.warning(f"NO LAB FOUND {current_file}")
+    def _build_details_dict(self, details_div: Tag) -> dict[str, Any]:
+        def parse_single_detail_entry(key, entry):
+            normalized_key = DETAILS_KEY_NORMALIZATION_DICT[key]
+            normalization_func = DETAILS_KEY_TO_NORMALIZATION_FUNCTION.get(normalized_key, None)
+            normalized_entry = (
+                FIPSHTMLParser.normalize_string(entry.text) if not normalization_func else normalization_func(entry)
+            )
+            return normalized_key, normalized_entry
+
+        entries = details_div.find_all("div", class_="row padrow")
+        entries = zip(
+            [x.find("div", class_="col-md-3") for x in entries], [x.find("div", class_="col-md-9") for x in entries]
+        )
+        entries = [(FIPSHTMLParser.normalize_string(key.text), entry) for key, entry in entries]
+        entries = [parse_single_detail_entry(*x) for x in entries if x[0] in DETAILS_KEY_NORMALIZATION_DICT.keys()]
```

Comment:
Hmm, we are losing some information here right? Or rather, that the check that `x[0] in DETAILS_KEY_NORMALIZATION_DICT.keys()` is required means that there is some field type on some cert page that is not in the `DETAILS_KEY_NORMALIZATION_DICT`. I think instead of the check we should do a run and log all of the additional details (along with their values) and then find a way to parse them as well.

Reply:
It's not difficult to parse additional info it we miss some (and we do). The question is to decide whether what we miss is worth parsing and storing. ATM we miss `Allowed Algorithms` which I don't find useful personally. 

I probably don't want to parse it all without some benefit analysis. I've prepared an issue: #289